pkg/kubectl/sorting_printer.go

```diff
@@ -60,8 +61,35 @@ func (s *SortingPrinter) sortObj(obj runtime.Object) error {
 	if len(objs) == 0 {
 		return nil
 	}
+
+	sorter, err := SortObjects(objs, s.SortField)
+	if err != nil {
+		return err
+	}
+
+	switch list := obj.(type) {
+	case *v1.List:
```

Comment:
As far as a I can tell, this comes directly from the Builder on a `kubectl get` I agree it seemed odd to me that was what I was seeing, but that's what I see on `kubectl get pods -o=name --sort-by=.metadata.name`